### DISCUSSION TITLE
Vigotec vg-l7x

### DISCUSSION THREAD

**Discussion by MIGUEL376** [arkypita/LaserGRBL#1305] [General]
Hello everyone! I'm new here. In the vigo works software I can view the temperature reading, how can I add this function in GRBL 4.1.0? I'm very happy with GRBL, congratulations on the work, it's helping me a lot.
![VigoWorks v3 1](https://user-images.githubusercontent.com/77809368/105736605-bde74600-5f13-11eb-8384-c294d5170678.jpg)

**Comment by arkypita** [comment]
Since there is no physical temperature probe mounted on the laser head, I deduce that the indicated temperature is that of the wifi chip on the control board. So totally useless.

Do you have a way to verify it?

**Reply by MIGUEL376** [reply]
First you want to thank for the answer ... I did some tests here putting ice on the laser module, the temperature dropped a lot. So I deduced that the temperature sensor is in the module. Observe the blue wire leaves the module and goes to the controller board.
 Thanks!
![20210126_134640](https://user-images.githubusercontent.com/77809368/105876952-7f838100-6010-11eb-9422-56da9aa267c8.jpg)
![20210126_134640](https://user-images.githubusercontent.com/77809368/105876993-8a3e1600-6010-11eb-871e-1a59c44b6fb4.jpg)

**Reply by arkypita** [reply]
Oh, ok... well done, good catch.
I don't own a Vigo, not even the control board. I had one on loan (just the control board) so I didn't notice this thing.

You must know that vigo boards use their own way of talking to their application, I don't know if you've read
https://github.com/arkypita/Vigotech-VG-L7X

It is necessary to understand how this data is communicated, if it is clear (or encoded) and how to possibly convert the raw data into a temperature.

I have dumped some communication, and I assume it is transmitted in the LTC field.
https://github.com/arkypita/Vigotech-VG-L7X/tree/main/Protocol

Since I made it go without the cable connected, I always read 4095 which I think is the end of scale.

You should make me an analysis of the communication and see which values it reports, related to the different temperatures you see.

You can do it by using eltma serial port monitor: https://www.eltima.com/it/products/serial-port-monitor/

**Reply by arkypita** [reply]
otherwise I can't help you

**Comment by MIGUEL376** [comment]
Tutto bene ... although I have grandparents and parents all Italians,
I didn't learn the language ... shame kkk.
I downloaded the eltima and did the analysis, I don't know if I did it right.

attachment




Em ter., 26 de jan. de 2021 às 21:38, arkypita <notifications@github.com>
escreveu:

**Reply by arkypita** [reply]
attachment by email does not work in github

you should go to the site and upload the file: https://github.com/arkypita/LaserGRBL/discussions/1305#discussioncomment-320773

**Reply by MIGUEL376** [reply]
In session one, the temperature was 26.3 degrees, oscillating a little.
At two it started at 26.3 and went up to 30.2.
This is more or less because it is too fast to set the right number.
[Monitoring Session1.txt](https://github.com/arkypita/LaserGRBL/files/5896572/Monitoring.Session1.txt)
[Monitoring Session2.txt](https://github.com/arkypita/LaserGRBL/files/5896588/Monitoring.Session2.txt)

**Reply by MIGUEL376** [reply]
That's right that you asked me? 
If I'm wrong, tell me how to set up the serial port monitor, and it's a bit complicated for me. 
Thank you very much.

**Reply by arkypita** [reply]
It's ok, but i am quite busy this time. In particular I am following the development of a firmware for vigo that allows to use it without hardware modifications.

**Comment by MIGUEL376** [comment]
wow this is great ... I don't understand much, but if I hear anything I can
help with, I am available.

Em ter., 2 de fev. de 2021 às 14:06, arkypita <notifications@github.com>
escreveu:
